### PR TITLE
ASTBuilder 

### PR DESCRIPTION
**Pull request by JakubSchwenkbeck**
### Motivation

This PR reworked and implemented the core functionality of the AST generator

### Changes 

- `ANTLR-grammar` :  added Object- and ArrayCreation, reworked primary to allow changing method calls via `new` and fixed class definition errors
- `AST` : I added several new Statements and Expressions to correctly reflect out java subset
- `AST Builder ` :  With a Top-Down approach completly reworked the ASTBuilder. This included all visitors for our current (rather complex) test-file and building into our AST strucutre. 

### Testing and further doing:
I will go on and refactor or add small edge cases to the ASTBuilder but the core functionality should be set. I would be happy if we add more test cases (which might bring the current ASTbuilder to crashing) so I can see what parts of our Java Subset I haven't handled yet.

### CODE REVIEW

File: src/main/java/de/students/Parser/AST.scala [hunk 1]
```diff
@@ -10,13 +10,16 @@ case class Program(classes: List[ClassDecl]) extends ASTNode
 case class ClassDecl(
                       name: String,
                       parent: Option[String],
+                      isAbstract : Boolean,
```
Comment (Brofian) [inline src/main/java/de/students/Parser/AST.scala:13]: Small comment to the naming convention of properties:
In many cases, it is recommended to keep the property names at their core. This for example would be `abstract: Boolean`. The reason is the distinction between the property and any possible getter method, that would often be called isAbstract (for booleans, instead of getAbstract) :)

File: src/main/java/de/students/Parser/ASTBuilder.scala [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+
+    override def visitAttribute(ctx: AttributeContext): VarDecl = {
+      val name = ctx.IDENTIFIER().getText
+      val varType = visitMyType(ctx.`type`())
+      VarDecl(name, varType)
+    }
+
+     def visitMyType(ctx: TypeContext): Type = {
```
Comment (Brofian) [inline src/main/java/de/students/Parser/ASTBuilder.scala:58]: Where exactly, does the `My` come from? Terms like these are not really verbose about their meaning of why they are there in the first place :)
https://youtu.be/-i9xDEpCVPI?t=220 ;)

**Comment by Brofian** [comment]
Looking in general very good by now :+1: :)  
I think it's up to you, if you want to keep this as a draft or merge your progress. We are a small enough team with mostly clearly defined work areas on a new private project. There is no reason to ensure a working state at all time ^^

**Review by Brofian — APPROVED** [review]
